plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/core/explorer/cwqTab/CwQTreeRootNode.kt

```diff
@@ -5,22 +5,57 @@ package software.aws.toolkits.jetbrains.core.explorer.cwqTab
 
 import com.intellij.ide.projectView.PresentationData
 import com.intellij.ide.util.treeView.AbstractTreeNode
-import com.intellij.openapi.extensions.ExtensionPointName
+import com.intellij.openapi.actionSystem.ActionGroup
+import com.intellij.openapi.actionSystem.ActionManager
 import com.intellij.openapi.project.Project
-import software.aws.toolkits.jetbrains.core.explorer.cwqTab.nodes.CwQServiceNode
+import software.aws.toolkits.jetbrains.ToolkitPlaces
+import software.aws.toolkits.jetbrains.core.credentials.AwsBearerTokenConnection
+import software.aws.toolkits.jetbrains.core.credentials.ToolkitAuthManager
+import software.aws.toolkits.jetbrains.core.explorer.actions.AnActionTreeNode
+import software.aws.toolkits.jetbrains.core.explorer.cwqTab.nodes.QRootNode
+import software.aws.toolkits.jetbrains.core.gettingstarted.editor.ActiveConnection
+import software.aws.toolkits.jetbrains.core.gettingstarted.editor.ActiveConnectionType
+import software.aws.toolkits.jetbrains.core.gettingstarted.editor.BearerTokenFeatureSet
+import software.aws.toolkits.jetbrains.core.gettingstarted.editor.checkBearerConnectionValidity
+import software.aws.toolkits.jetbrains.services.codemodernizer.explorer.nodes.CodeModernizerRunModernizeNode
+import software.aws.toolkits.jetbrains.services.codemodernizer.isCodeModernizerAvailable
 
 class CwQTreeRootNode(private val nodeProject: Project) : AbstractTreeNode<Any>(nodeProject, Object()) {
+    private val runCodeModernizerNode by lazy { CodeModernizerRunModernizeNode(nodeProject) }
+
     override fun update(presentation: PresentationData) {}
 
-    override fun getChildren(): Collection<AbstractTreeNode<*>> = EP_NAME.extensionList
-        .filter { it.enabled() }
-        .map {
-            it.buildServiceRootNode(nodeProject).also { node ->
-                node.parent = this
+    override fun getChildren(): Collection<AbstractTreeNode<*>> {
+        val connection = checkBearerConnectionValidity(project, BearerTokenFeatureSet.Q)
+
+        val groupId = when (connection) {
+            is ActiveConnection.NotConnected -> if (otherActiveIdentityConnectionsAvailable()) ENABLE_Q_ACTION_GROUP else Q_SIGNED_OUT_ACTION_GROUP
+            is ActiveConnection.ValidBearer -> Q_SIGNED_IN_ACTION_GROUP
+            else -> Q_EXPIRED_TOKEN_ACTION_GROUP
+        }
+
+        val actions = ActionManager.getInstance().getAction(groupId) as ActionGroup
+        val childNodes = actions.getChildren(null).mapNotNull {
+            AnActionTreeNode(project, ToolkitPlaces.CWQ_TOOL_WINDOW, it)
+        }
+
+        // Temporary put the transform node in the panel, to be removed soon.
+        if (groupId == QRootNode.Q_SIGNED_IN_ACTION_GROUP) {
+            if (connection.connectionType == ActiveConnectionType.IAM_IDC && isCodeModernizerAvailable(project)) {
```

Comment:
`isCodeModernizerAvailable` already checks IDC so remove this redundant first part of the IF

Reply:
removed in the next commit.